src/components/Transactions/TransactionDetail.tsx

```diff
@@ -189,60 +189,77 @@ export default function TransactionDetail({
               pb={4}
               ml={{ md: 3 }}
               pr={3}>
-              <Stack>
+              <Stack mb={3}>
                 <Typography fontWeight={700} variant="caption">
                   {t('app.transactionDetail.card2.title').toUpperCase()}
                 </Typography>
               </Stack>
-              <Grid item xs={12} sm={12} pt={3}>
-                <Typography sx={{ wordBreak: 'break-word' }} color={theme.palette.text.secondary}>
-                  {t('app.transactionDetail.paidBy')}
-                </Typography>
-                <Stack
-                  direction={{ xs: 'column', sm: 'row', md: 'column' }}
-                  spacing={{ xs: 0, sm: 1, md: 0 }}>
-                  <Typography sx={{ wordBreak: 'break-word' }} fontSize={16} fontWeight={600}>
-                    {transactionData.payer.name}
+
+              {transactionData.payer.name !== '-' && (
```

Comment:
We could make these fields optionals, and instead of returning them as '-' when not present and checking if their value is '-', we could just avoid returning the payer object, (or returning it null)  and rendering the whole block (payer info and payment method info) based on the existence of the payer object. 

What do you think? I think this makes the code cleaner.

Reply:
I agree, did it.